File: src/main/java/cn/zjiali/robot/main/OutMessageConvert.java
package cn.zjiali.robot.main;

import cn.zjiali.robot.util.PluginConfigUtil;
import cn.zjiali.robot.model.message.OutMessage;
import cn.zjiali.robot.util.MessageUtil;

/**
 * 发送消息转换类
 *
 * @author zJiaLi
 * @since 2021-09-04 11:26
 */
public class OutMessageConvert {

    private static final OutMessageConvert instance = new OutMessageConvert();

    public static OutMessageConvert getInstance() {
        return instance;
    }

    public String convert(OutMessage outMessage) {
        if (outMessage == null) return null;
        String content = outMessage.getContent();
        if (!outMessage.isConvertFlag()) return content;
        String templateCode = outMessage.getTemplateCode();
        String template = PluginConfigUtil.getTemplate(templateCode);
        int fillFlag = outMessage.getFillFlag();
        String finalMessage = MessageUtil.replaceMessage(template, fillFlag == 1 ? outMessage.getFillMap() : outMessage.getFillObj());
        outMessage.setFinalMessage(finalMessage);
        return finalMessage;
    }

}

File: src/main/java/cn/zjiali/robot/main/OutMessageConvert.kt
package cn.zjiali.robot.main

import cn.zjiali.robot.config.AppConfig
import cn.zjiali.robot.constant.AppConstants
import cn.zjiali.robot.manager.PluginManager
import cn.zjiali.robot.model.message.OutMessage
import cn.zjiali.robot.util.GuiceUtil
import cn.zjiali.robot.util.MessageUtil
import cn.zjiali.robot.util.PluginConfigUtil

/**
 * 发送消息转换类
 *
 * @author zJiaLi
 * @since 2021-09-04 11:26
 */
class OutMessageConvert {

    fun convert(outMessage: OutMessage?): String? {
        if (outMessage == null) return null
        val content = outMessage.content
        if (!outMessage.isConvertFlag) {
            outMessage.finalMessage = content
            return content
        }
        val templateCode = outMessage.templateCode
        val pluginManager = GuiceUtil.getBean(PluginManager::class.java)
        val template =
            if (outMessage.fromMsgType == AppConstants.MSG_FROM_GROUP && AppConfig.serverControl())
                pluginManager.getTemplate(outMessage.pluginCode, templateCode, outMessage.groupId, outMessage.senderId)
            else PluginConfigUtil.getTemplate(outMessage.pluginCode, templateCode)
        val fillFlag = outMessage.fillFlag
        val finalMessage = if (fillFlag == AppConstants.FILL_OUT_MESSAGE_OBJECT_FLAG) template.let {
            MessageUtil.replaceMessage(
                it,
                outMessage.fillObj
            )
        }
        else template.let { MessageUtil.replaceMessageByMap(it, outMessage.fillMap) }
        outMessage.finalMessage = finalMessage
        return finalMessage
    }

    companion object {
        @JvmStatic
        val instance = OutMessageConvert()
    }
}